face/unix-stream-transport.go

```diff
@@ -8,36 +8,41 @@
 package face
 
 import (
+	"bufio"
+	"io"
 	"net"
 	"runtime"
 	"strconv"
 
 	"github.com/named-data/YaNFD/core"
 	"github.com/named-data/YaNFD/face/impl"
-	"github.com/named-data/YaNFD/ndn"
-	"github.com/named-data/YaNFD/ndn/tlv"
+	ndn_defn "github.com/named-data/YaNFD/ndn_defn"
+
+	enc "github.com/zjkmxy/go-ndn/pkg/encoding"
 )
 
 // UnixStreamTransport is a Unix stream transport for communicating with local applications.
 type UnixStreamTransport struct {
-	conn *net.UnixConn
+	conn   *net.UnixConn
+	reader *bufio.Reader
 	transportBase
 }
 
 // MakeUnixStreamTransport creates a Unix stream transport.
-func MakeUnixStreamTransport(remoteURI *ndn.URI, localURI *ndn.URI, conn net.Conn) (*UnixStreamTransport, error) {
+func MakeUnixStreamTransport(remoteURI *ndn_defn.URI, localURI *ndn_defn.URI, conn net.Conn) (*UnixStreamTransport, error) {
 	// Validate URIs
 	if !remoteURI.IsCanonical() || remoteURI.Scheme() != "fd" || !localURI.IsCanonical() || localURI.Scheme() != "unix" {
 		return nil, core.ErrNotCanonical
 	}
 
 	t := new(UnixStreamTransport)
-	t.makeTransportBase(remoteURI, localURI, PersistencyPersistent, ndn.Local, ndn.PointToPoint, tlv.MaxNDNPacketSize)
+	t.makeTransportBase(remoteURI, localURI, PersistencyPersistent, ndn_defn.Local, ndn_defn.PointToPoint, ndn_defn.MaxNDNPacketSize)
 
 	// Set connection
 	t.conn = conn.(*net.UnixConn)
+	t.reader = bufio.NewReaderSize(t.conn, 32*ndn_defn.MaxNDNPacketSize)
```

Comment:
How you decided `32*ndn_defn.MaxNDNPacketSize`? Just want something large enough?

Reply:
Yeah it's arbitrary. It's a *minimum* anyway, the buffer could be larger